examples/wave.py

```diff
@@ -256,14 +275,23 @@ def rhs(t, w):
     parser.add_argument("--casename", help="casename to use for i/o")
     parser.add_argument("--numpy", action="store_true",
         help="use numpy-based eager actx.")
+    parser.add_argument("--nompi", action="store_true", help="Disable MPI")
     args = parser.parse_args()
     casename = args.casename or "wave"
+
     from mirgecom.array_context import get_reasonable_array_context_class
     actx_class = get_reasonable_array_context_class(lazy=args.lazy,
-                                                    distributed=True,
+                                                    distributed=not args.nompi,
                                                     profiling=args.profiling,
                                                     numpy=args.numpy)
 
-    main(actx_class, use_logmgr=args.log, casename=casename)
+    if args.nompi:
+        import inspect
+        # run main without the mpi_entry_point wrapper
+        inspect.unwrap(main)(
+            actx_class, use_logmgr=args.log, casename=casename,
+            mpi=not args.nompi)
+    else:
+        main(actx_class, use_logmgr=args.log, casename=casename, mpi=not args.nompi)
```

Comment:
I think you can do this with:

```suggestion
    if args.mpi:
        main_func = mpi_entry_point(main)
    else:
        main_func = main

    main_func(actx_class, use_logmgr=args.log, casename=casename, mpi=args.mpi)
```

(and removing the decorator on `main`).